src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -1063,5 +1080,59 @@ private void EnsureMethodCanBeInvoked(string methodName)
                 throw new InvalidOperationException(Resources.FormatRazorPage_MethodCannotBeCalled(methodName, Path));
             }
         }
+
+        private struct AttributeInfo
```

Comment:
Seems wasteful to have this attribute info reflect TagHelper and non-TagHelper attributes. The TagHelper variation would be much simpler if they weren't combined.